[Alientian] iP

alientian opened the pull request:
# ET
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

ET frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use
All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉 
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

alientian committed:
Rename, greet, exit

alientian committed:
Rename, greet, exit

alientian committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Duke.java

alientian committed:
echo

alientian committed:
Add, List

alientian committed:
add Mark as done

alientian committed:
Add toString to Task.java

alientian committed:
Add ToDos, Events, Deadlines files

alientian committed:
Add Automated Text UI Test Case

alientian committed:
Add handle errors

alientian committed:
Add Delete

alientian committed:
Add Save

alientian committed:
Merge branch 'branch-Level-7'

alientian committed:
Add Dates

alientian committed:
Merge branch 'branch-Level-8'

alientian committed:
Add data directory

alientian committed:
Add Ui, Storage, Parser, TaskList, Command class

alientian committed:
Add Ui, Storage, Parser, TaskList, Command class

alientian committed:
Merge remote-tracking branch 'origin/master'

alientian committed:
Move files to duke package

alientian committed:
Merge remote-tracking branch 'origin/add-gradle-support'

alientian committed:
Run Duke using gradle

alientian committed:
Add JUnit Tests

alientian committed:
Add JavaDoc comments

alientian committed:
Update code to comply coding standard

alientian committed:
Add Find

alientian committed:
Merge branch 'branch-A-JavaDoc'

alientian committed:
Resolve merge conflict

alientian committed:
Add JavaDoc for Find

alientian committed:
branch-Level-9: Resolve merge conflict

kimshitong commented on src/main/java/duke/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Exits duke and saves task list to storage
+ */
+public class ExitCommand extends Command{
```
Minor standard mistake on space between your function and open curly braces.

`public class ExitCommand extends Command {`

kimshitong commented on src/main/java/duke/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+    public ExitCommand() {}
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException{
```
Minor standard mistake on space between your function and open curly braces.

`    public void execute(TaskList tasks, Ui ui, Storage storage) throws IOException {`

kimshitong commented on src/main/java/duke/ListCommand.java:
```diff
@@ -0,0 +1,21 @@
+/**
+ * Lists all the tasks in task list
+ */
+public class ListCommand extends Command{
```
Minor standard mistake on space between your function and open curly braces.

`public class ListCommand extends Command {`

kimshitong commented on src/main/java/duke/MarkCommand.java:
```diff
@@ -0,0 +1,25 @@
+public class MarkCommand extends Command{
+
+    /** Index of the task in task list */
+    private final int INDEX;
```
Good Use on Capitalising final variable

kimshitong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+     * @return the task list.
+     * @throws IOException if there were errors reading and/or converting data from file.
+     */
+    public ArrayList<Task> load() throws IOException{
```
similar, spaces between your Exception and open bracket

kimshitong commented on src/main/java/duke/Ui.java:
I think you can use a final variable horizontalLine = "____________________________________________________________" to maintain consistency and improve readability of your code

kimshitong commented on src/main/java/duke/UnmarkCommand.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * Marks a task in task list as not done
+ */
+public class UnmarkCommand extends Command{
```
similar, space between function and {

ldinghan commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,23 @@
+     * Checks if command is exit command
+     * @return true if it is exit command
+     */
+    public abstract boolean isExit();
```
Not sure if this is the right approach, considering how there is only one exit command (and almost all of the commands will return false)

ldinghan commented on src/main/java/duke/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+public class DeleteCommand extends Command{
+
+    /** Index of the task in task list */
+    private final int INDEX;
```
Saw that your commands used the variable name `INDEX`, perhaps it could be more specific? For example `INPUT_INDEX` or `TASK_INDEX` or something like that, which gives a bit more information about what this variable is about

ldinghan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,139 @@
+/**
+ * Text UI of the application.
+ */
+public class Ui {
```
Really liked the well-chosen method names in your Ui class!

ldinghan commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString()
```
Might want to use `String.format(...)` to make this a little easier to read!

ldinghan reviewed (commented):
Looks good! Names were well chosen, especially the Ui methods. Follows the Code Quality guide well!

kimshitong commented on src/main/java/duke/AddCommand.java:
I am thinking that would it be possible you to transform the lengthy operations into a method so that it improves the readability.

```
            } else if (TASK_TO_ADD.startsWith("deadline ")) {
                String[] words = TASK_TO_ADD.split("/");
                if (words.length != 2 || words[0].length() < 10 || words[1].length() < 4) {
                    throw new IncompleteInputException("deadline");
                }
                String des = words[0].substring(9);
                String date = words[1].substring(3);
                LocalDate by = LocalDate.parse(date);
                Deadline dl = new Deadline(des, by);
                tasks.add(dl);
                ui.showTaskAdded(dl, tasks.total());


            } else if (TASK_TO_ADD.startsWith("event ")) {
```

where you can transform below 

```
                String des = words[0].substring(9);
                String date = words[1].substring(3);
                LocalDate by = LocalDate.parse(date);
                Deadline dl = new Deadline(des, by);
                tasks.add(dl);
```

become some method like

Deadline dl = unparseDeadline(words);

so it is something like

```
            } else if (TASK_TO_ADD.startsWith("deadline ")) {
                String[] words = TASK_TO_ADD.split("/");
                if (words.length != 2 || words[0].length() < 10 || words[1].length() < 4) {
                    throw new IncompleteInputException("deadline");
                }
                tasks.add(unparseDeadline(words)); // Sorry for bad naming, cant come up something. maybe can name it as transformDeadline or smth
                ui.showTaskAdded(dl, tasks.total());


            } else if (TASK_TO_ADD.startsWith("event ")) {
```

kimshitong reviewed (commented):
Overall Looks Good!

alientian committed:
Add GUI using JavaFX

alientian committed:
Change images

alientian committed:
Change to round displayImage

alientian committed:
Edit UI responses

alientian committed:
Merge branch 'branch-Level-10'

alientian committed:
Improve code quality

AddCommand: Remove complicated expressions to improve readability of code

Delete unused code

alientian committed:
Merge pull request #1 from alientian/branch-A-CodeQuality

Improve code quality

alientian committed:
Add assertions

Current does not use assertions

Assert feature is useful for documenting important assumptions that should hold at various points in the code.

Lets add assert feature in Parser, Storage and Task classes.

alientian committed:
Merge pull request #2 from alientian/branch-A-Assertions

Add assertions

alientian committed:
Find command: make matching case-insensitive

Find command is case-sensitive.

A case-insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

Let's update the search algorithm to use case-insensitive matching.

alientian committed:
Merge branch 'branch-C-BetterSearch'

alientian committed:
Add show welcome message

alientian committed:
Merge branch 'branch-A-BetterGui'

alientian committed:
Update README: add User Guide

alientian committed:
Edit typo errors

alientian committed:
Add hyperlinks in README

alientian committed:
Add Ui.png

alientian committed:
Edit errors

alientian committed:
Add exit programme

alientian committed:
Merge branch 'branch-A-BetterGui'

alientian committed:
Edit bugs to release jar
